worker/manager/src/mirrors_qa_manager/worker.py

```diff
@@ -0,0 +1,309 @@
+
+    def run(self) -> None:
+        logger.info("Starting worker manager.")
+        try:
+            # Ensure the wireguard container is still up
+            exec_command(self.wg_container, self.wg_healthcheck_cmd)
+            while True:
+                tests = self.fetch_tests()
+                for test in tests:
+                    test_id = test["id"]
+                    country_code = test["country_code"]
+                    # Fetch the configuration file for the requested country
+                    try:
+                        self.copy_wireguard_conf_file(country_code)
+                    except FileNotFoundError:
+                        logger.warning(
+                            f"Could not find {country_code}.conf for "
+                            f"{test_id}. Skipping."
+                        )
+                        continue
+                    except Exception:
+                        logger.exception(
+                            f"error while fetching {country_code}.conf for {test_id}"
+                        )
+                        continue
+
+                    logger.info(
+                        f"Reconfiguring wireguard network interface for {country_code}"
+                    )
+
+                    # After copying the file, restart the interface.
+                    exec_command(self.wg_container, self.wg_down_cmd)
+                    exec_command(self.wg_container, self.wg_up_cmd)
+                    # Perform another healthcheck to ensure traffic can go
+                    # through.
+                    # TODO: Use the result from the healthcheck call to
+                    # populate the IP-related data as the task container
+                    # doesn't know anything about its network.
+                    # Could also be used to validate that the country config
+                    # is actually for this country in case the 'host'
+                    # wrongly names a config file.
+                    exec_command(self.wg_container, self.wg_healthcheck_cmd)
+
+                    logger.info(f"Processing test {test_id}")
+                    # Start container for the task
+                    output_filename = f"{test_id}.json"
+                    self.start_task_container(
+                        Settings.TASK_WORKER_IMAGE, output_filename=output_filename
+                    )
+                    output_fpath = self.instance_dir / output_filename
+                    results = output_fpath.read_bytes()
+                    logger.info(f"Got results from test {test_id}: {results}")
+                    # TODO: Merge the IP data from the healthcheck cmd and the resutls
+                    # an dupload to the Backend API
+                    logger.info(f"Uploading results for {test_id}")
+                    output_fpath.unlink()
+                self.sleep()
+        except Exception as exc:
+            logger.exception("error while processing tasks")
```

Comment:
```suggestion
            logger.exception(f"error while processing tasks: {exc!s}")
```

Just in case exiting raises additional stuff ; core issue is printed first